src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -157,13 +159,20 @@ private static bool ShouldConsiderSymbol(ISymbol symbol)
 
             var additionalReferences = new List<Location>();
 
-            foreach (var document in documentsToSearch)
+            foreach (var additionalDocument in documentsToSearch)
```

Comment:
Nit: I don't love the name `additionalDocument`, since it implies that it's not part of `documentsToSearch`.  Maybe rename `document` to `startingDocument` and `additionalDocument` to `currentDocument`?